### ISSUE TITLE
User request: amps instead of milliamps

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/ohms-law#153]
> Could you allow the simulation to provide the current value in Amps, rather than milliAmps?  My students are middle schoolers who are taking high school physical science, and they have not really learned the metric system.  They always get confused when they set the voltage 0.1V and the resistance to 800 ohms and the simulation shows 0.1mA for current.  However, when they do the calculation themselves, 0.1V divided by 800 ohms, they get .000125A.  It is very difficult to explain to them how and why milli amps and amps are different.  Could you please make this change to where it shows current in Amps?

**Comment by ariel-phet** [comment]
@arouinfar @zepumph I think this might be a nice option to add, and it might be a good simple case of modifying a sim that has description (I believe). As we have more and more sims outfitted with various a11y features we will want to consider the process of updating sims with feature requests. I think this one falls on the fairly simple side.

I could think of 3 ways to accomplish such an option fairly easily:

1. Adding a query parameter and putting this in an options menu much like our projector mode or such (probably the easiest/most straightforward)

2. Using the "readout/combo box" hybrid used in something like gas properties - probably the cleanest in some ways, and presenting the option easily to the user, but likely tougher for a11y in some ways. mA as the default

3. Having a "units" radio button selection included in the UI, iwith mA as the default

@zepumph would there be a simple way to just change the description if this other option were chosen, I assume that is straightforward?

@arouinfar thoughts on the options above?

**Comment by zepumph** [comment]
> this other option were chosen

Do you mean (3)?

**Comment by zepumph** [comment]
I investigated this, and after looks into it more I thought I would experiment with option (3). Before I knew it I had just done it (about an hour of poking around). I committed it just to see what people think. It would be easy to tweak or revert, just let me know what is best.

Updating the descriptions were very simple, but depending on if we go with the radio buttons or not, we will need to add some labels and potentially help text.

**Comment by zepumph** [comment]
![image](https://user-images.githubusercontent.com/6856943/73034772-612de880-3df2-11ea-972c-d27a964c40ae.png)

**Comment by arouinfar** [comment]
I like the idea of supporting amps, and I think it may even be preferable for a lesson more focused on proportional reasoning since it would eliminate the need for unit conversion.

Of the ideas proposed by @ariel-phet, I most prefer options (1) or (3). I think the combobox would add more complexity than necessary to achieve the goal.

@zepumph thanks for getting option (3) up and running! I think the behavior is looking good, but I don't really love the addition of the radio group to the play area. Despite the reasonable alignment/placement the controls feel a bit awkward (maybe I'm just not used to it?). Flow-wise, I'm debating on whether or not we want these controls in the play area since it would encourage students to play with it. I don't think it would cause any harm, but I'm not sure it supports the learning goals. If others like this approach, I would be fine with it, though I would definitely reduce the font size a bit. Maybe 16 pt?

Personally, I would prefer to bury the A/mA controls in the Options menu, and potentially add the ability to set the units with a query parameter like `units=A` or `units=mA`. This would add more overhead to the implementation and would require more description work (another dialog & radio button group), but it doesn't seem unreasonable. 

Curious to hear @ariel-phet's thoughts here.

**Comment by ariel-phet** [comment]
@arouinfar I am not sure if I would bury the units in an options menu for this one. I could very easily see this being desirable for iO clients, and considering we have a teacher request, if we can accomplish the request in a way that we are happy with for flow and such, I think it is reasonable to include.

I think a good example is Under Pressure, which has a "Units" panel with radio buttons allowing various units to be selected (Metric vs English in that case)

But what about a layout like below, where the Units are not too emphasized (reduced font size, and mA first since that is the default), as well as reasonably aligned with the current readout:

![ohms](https://user-images.githubusercontent.com/4051174/73240110-3a0d4900-415b-11ea-9062-afd385546045.PNG)

**Comment by arouinfar** [comment]
> @arouinfar I am not sure if I would bury the units in an options menu for this one. I could very easily see this being desirable for iO clients, and considering we have a teacher request, if we can accomplish the request in a way that we are happy with for flow and such, I think it is reasonable to include.

Alright, I've been convinced. 😄 

I tinkered a bit more with the alignment of the radio group, and I think either of these options could work well.

(1) Center-aligned with current panel & left-aligned with V/R panel.
![image](https://user-images.githubusercontent.com/8419308/73288839-f1847880-41b8-11ea-8842-69ecfc48b1d8.png)

(2) Top-aligned with current panel & left-aligned with V/R panel.
![image](https://user-images.githubusercontent.com/8419308/73288849-f812f000-41b8-11ea-889b-f603a8aa17fc.png)

I have a slight preference for (2) because of the extra negative space, but I'll defer to @ariel-phet.

**Comment by ariel-phet** [comment]
@zepumph lets go with (2) in @arouinfar mockup above

**Comment by zepumph** [comment]
Implemented above. Please review for visual pixel polishing. Note that all display logic like font size is in `UnitsRadioButtonContainer`, and layout is controlled at the bottom of `OhmsLawScreenView` with the object called `unitsRadioButtonContainer`.

I think we should also loop @terracoda now that we have a direction, as there may need to be PDOM and alert changes. @terracoda can you please take a look at master, specifically the radio button group, and review my changes thus far (all current descriptions should update based on what the selected unit is) as well as recommending any further work needed. I know that alerts for when radio buttons change has been done in the past. Also Please review the structure of the PDOM, and the heading "Current Units".

Onward, to designers!

**Comment by terracoda** [comment]
@arouinfar and @ariel-phet, in the non-visual experience we have headings for Play Area and Control Area, so gien @arouinfar concern, I assume the radio buttons should go in the Control Area. For non-visual users that keeps them out of the Play Area.

I prefer layout (2) as well.

**One question:** Are the units just for the circuit or current or does it matter?
The heading could be "Units for Circuit" or "Units for Current" or I could get this idea across in the help text. @arouinfar and @ariel-phet, which help text do you prefer?

**H3: Units**
@arouinfar and @ariel-phet, which help text do you prefer?
- "Choose unit for current." or 
- "Choose unit for circuit."

**Labels for radio buttons are straight forward:**
- Milliamps (mA)
- Amps (A)

@zepumph, the object and/or context responses would need new parameters depending on the selected unit. 

I haven't tried it out yet.

**Comment by terracoda** [comment]
I checked the design doc, and of course values for current are only in dynamic descriptions and in context responses.

https://docs.google.com/spreadsheets/d/15LuKXlzkh6xHpNbWC4ujtOfz1F7FJ7tj4oUF8Jvmj-k/edit#gid=0

**Comment by terracoda** [comment]
In addition, we need to update the screen summary:

> Ohm's Law‬ is an interactive sim. It changes as you play with it. It has a Play Area and a Control Area.
> In the Play Area you find the equation for Ohm's Law, V equals I times R, and a circuit. Voltage and resistance sliders allow changes to the equation and circuit. In the Control Area, a button allows users to reset the sim.

**Add the new radio buttons in under the Control Area:**
> Ohm's Law‬ is an interactive sim. It changes as you play with it. It has a Play Area and a Control Area.
>
> In the Play Area you find the equation for Ohm's Law, V equals I times R, and a circuit. Voltage and resistance sliders allow changes to the equation and circuit. 
>
> **In the Control Area, radio buttons swap units for current and a button resets the sim.**

@zepumph, go ahead and change to 3 paras instead of two, if that's easy.

**Comment by terracoda** [comment]
@ariel-phet and @arouinfar, these text changes sound good to you? Please see previous comment and the help text choice in https://github.com/phetsims/ohms-law/issues/153#issuecomment-579830533

**Comment by arouinfar** [comment]
@zepumph the vertical alignment of the radio group is looking a bit off. If it's tricky to precisely align, perhaps you can manually shift the radio group down another few px?
![image](https://user-images.githubusercontent.com/8419308/73409245-f716b700-42bb-11ea-990f-b02cebd966b0.png)

@terracoda the H3 label, radio button labels, and screen summary all look great. As for help text, let's go with: "Choose unit for current."

**Comment by ariel-phet** [comment]
@terracoda 

- Yes these should be in the Control Area

- The units are for the current

- For help text I think "Choose units for current" or perhaps "Choose units for current display"

- I am not sure I love the phrase "swap units" but I think something more pedantic might just get long

**Comment by terracoda** [comment]
@ariel-phet, how about this? It's a bit longer, but nice and complete:
> Ohm's Law‬ is an interactive sim. It changes as you play with it. It has a Play Area and a Control Area.
>
>In the Play Area you find the equation for Ohm's Law, V equals I times R, and a circuit. Voltage and resistance sliders allow changes to the equation and circuit. 
>
> **The Control Area has radio buttons to switch between milliamps and amps, and a button to reset the sim.**

**Comment by ariel-phet** [comment]
@terracoda I think that is a nice improvement! I would go with that.

**Comment by terracoda** [comment]
Thanks @ariel-phet! @zepumph, I'll post a summary in case you need it after my meetings today.

**Comment by terracoda** [comment]
@zepumph, so to summarize the decisions from https://github.com/phetsims/ohms-law/issues/153#issuecomment-579830533 and https://github.com/phetsims/ohms-law/issues/153#issuecomment-580265556

The agreed upon State descriptions are:

In the screen/scene summary (bolded text):
> Ohm's Law‬ is an interactive sim. It changes as you play with it. It has a Play Area and a Control Area.
>
>In the Play Area you find the equation for Ohm's Law, V equals I times R, and a circuit. Voltage and resistance sliders allow changes to the equation and circuit. 
>
> **The Control Area has radio buttons to switch between milliamps and amps, and a button to reset the sim.**

New stuff for the Control Area:
**H3: Units**
- Paragraph of help text for radio buttons: 
  - Choose unit for current. 
- Labels for radio buttons:
  - Milliamps (mA)
  - Amps (A)

**Comment by zepumph** [comment]
>@zepumph the vertical alignment of the radio group is looking a bit off. If it's tricky to precisely align, perhaps you can manually shift the radio group down another few px?

I originally had just aligned their centers, it is hard to know the value of the top of the readout, so I ended up just shifting it manually a few pixels, how is that?

@terracoda I implemented your description design. Please review.

**Comment by arouinfar** [comment]
@zepumph looks perfectly aligned, thanks!

**Comment by samreid** [comment]
@zepumph asks if we would like to republish this soon.
@ariel-phet replies that it would be nice to do on the soon side, since it is not too complicated and came from a user.
@zepumph asks if it needs PhET-iO redesign?
@ariel-phet requests @zepumph to check in with @kathy-phet about whether we need to update the PhET-iO API.

**Comment by terracoda** [comment]
@zepumph, the descriptions sound fine, except there the first item in the Units list is counted twice. This might be a quirk to VoiceOver.
![Screen Shot 2020-02-07 at 1 36 46 PM](https://user-images.githubusercontent.com/1902116/74050382-8739b800-49b0-11ea-8b82-a667d5d1fb3b.png)

I also noticed a coding error. There is no parent `ul` Slider Controls, though there are `li`'s on each of the sliders. We didn't use lists for the sliders in GFL and GFLB. I don't remember why? I am not sure if we should add the parent `ul` or remove the `li`'s. Is there any documentation on why we removed the list structure in GFL and GFLB, was there a role conflict?

**Comment by terracoda** [comment]
And here is a screen shot of the code:
![Screen Shot 2020-02-07 at 1 46 45 PM](https://user-images.githubusercontent.com/1902116/74051009-bac91200-49b1-11ea-9635-122c5a1204e7.png)

**Comment by terracoda** [comment]
Also, @zepumph, could you associate the Units `ul` with the Units `h3` with an `aria-labelledby`? I realize that our group-naming conventions are not so consistent, I don't think we used `aria-labelledby` on the "Force Values" heading in GFL regular, but maybe we should. I'll make an issue for that.

**ALSO**, just realized (and am able to reproduce regularly) that the double counting that I hear in VO only happens when I am reading the list items backwards. This also happens in GFL on the Force Value radiobuttons, so I think this is definitely a VO bug. I'll make a note in the GFL issue about that.

**ALSO**, I am finding some focus inconsistencies on the list of slider controls, so I think it may be best to remove the list structure all together around the sliders and also remove the` aria-labelledby` that creates the group semantics as we did in GFL and GFLB. I think there was some weird issues with the sliders and lists and group semantics. Let me if your memory is different, I am wrong about this. 

And I need to make issue to document this some where! Maybe in Binder?

**Comment by terracoda** [comment]
@zepumph, I am going to create new issues.

**Comment by zepumph** [comment]
>ALSO, just realized (and am able to reproduce regularly) that the double counting that I hear in VO only happens when I am reading the list items backwards. This also happens in GFL on the Force Value radiobuttons, so I think this is definitely a VO bug. I'll make a note in the GFL issue about that.

I wonder if we shouldn't make a totally separate issue to investigate that?

**Comment by terracoda** [comment]
We could. I connected this issue to the GFL one.

**Comment by zepumph** [comment]
@terracoda I have done the breakout issues you have created. Besides those, is there anything else here?

**Comment by zepumph** [comment]
Pinging here for a review before moving further for this sim.

**Comment by terracoda** [comment]
The sliders and the radio groups sound great. The pdom structure looks good!
Thanks @zepumph!

I think this issue is done.

**Comment by terracoda** [comment]
@zepumph back to you for closing.

**Comment by arouinfar** [comment]
This was a great change! Looks like it will get picked up in the upcoming 1.5 release. Since it stems from a user request, I'm going to reopen so we remember to have @oliver-phet notify the user when the new version is published.
